ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/ActivePeers.hs

```diff
@@ -209,29 +214,32 @@ jobPromoteWarmPeer PeerSelectionActions{peerStateActions = PeerStateActions {act
   where
     handler :: SomeException -> Completion m peeraddr peerconn
     handler e =
-      --TODO: decide what happens if promotion fails, do we stay warm or go to
-      -- cold? Will this be reported asynchronously via the state monitoring?
+      -- Incase of an error we add a re-promotion delay and let monitoring
+      -- deal with setting the peer as cold if needed.
       Completion $ \st@PeerSelectionState {
                                activePeers,
+                               establishedPeers,
                                targets = PeerSelectionTargets {
                                            targetNumberOfActivePeers
                                          }
                              }
-                    _now -> Decision {
+                    now -> Decision {
         decisionTrace = TracePromoteWarmFailed targetNumberOfActivePeers
                                                (Set.size activePeers) 
                                                peeraddr e,
         decisionState = st {
                           inProgressPromoteWarm = Set.delete peeraddr
-                                                    (inProgressPromoteWarm st)
+                                                    (inProgressPromoteWarm st),
+                          establishedPeers = EstablishedPeers.setActivateTime
+                                               (Set.singleton peeraddr)
+                                               (rePromotionDelay `addTime` now)
+                                               establishedPeers
```

Comment:
Yeah ok, this seems like a reasonable solution. So we still leave it as warm, but avoid re-promoting again immediately.